src/dex/balancer-v1/balancer-v1.ts

```diff
@@ -0,0 +1,729 @@
+    return this.eventPools.allPools
+      .filter(
+        p =>
+          p.tokens.some(
+            token => token.address.toLowerCase() === from.address.toLowerCase(),
+          ) &&
+          p.tokens.some(
+            token => token.address.toLowerCase() === to.address.toLowerCase(),
+          ),
+      )
+      .slice(0, limit);
+  }
+
+  async getPoolPrices(
+    amounts: bigint[],
+    side: SwapSide,
+    unitVolume: bigint,
+    exchangeProxy: Address,
+    pool: OldPool | null,
+  ): Promise<PoolPrices<BalancerV1Data> | null> {
+    if (!pool) return null;
+    try {
+      let unit = await this.eventPools.getPoolPrices(pool, side, unitVolume);
+
+      const _width = Math.floor((amounts.length - 1) / BALANCER_CHUNKS);
+      const _amounts = Array.from(Array(BALANCER_CHUNKS).keys()).map(
+        i => amounts[(i + 1) * _width],
+      );
+
+      const _prices = await Promise.all(
+        _amounts.map(a => this.eventPools.getPoolPrices(pool, side, a)),
+      );
+
+      const prices = interpolate(_amounts, _prices, amounts, side);
+
+      return {
+        prices,
+        unit,
+        data: {
+          pool: pool.id,
+          exchangeProxy,
+        },
+        poolAddresses: [pool.id],
+        exchange: this.dexKey,
+        poolIdentifier: BalancerV1.getIdentifier(this.dexKey, pool.id),
+        gasCost: BALANCER_SWAP_GAS_COST,
+      };
+    } catch (e) {
+      this.logger.error(`Error_getPoolPrices: ${pool.id}`, e);
+      return null;
+    }
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<BalancerV1Data>> {
+    try {
+      const _from = wrapETH(srcToken, this.network);
+      const _to = wrapETH(destToken, this.network);
+
+      const allPools = this.getPools(_from, _to);
+      const allowedPools = limitPools
+        ? allPools.filter(({ id }) =>
+            limitPools.includes(BalancerV1.getIdentifier(this.dexKey, id)),
+          )
+        : allPools;
+
+      if (!allowedPools.length) return null;
+
+      const unitVolume = BigInt(
+        10 ** (side === SwapSide.SELL ? _from : _to).decimals,
+      );
+
+      const poolStates = this.eventPools.getState(blockNumber);
+      if (!poolStates) {
+        this.logger.error(`getState returned null`);
+        return null;
+      }
+
+      const missingPools = allowedPools.filter(
+        pool => !(pool.id.toLowerCase() in poolStates),
+      );
+
+      const missingPoolStates = missingPools.length
+        ? await this.eventPools.getAllPoolDataOnChain(
+            { pools: missingPools },
+            blockNumber,
+          )
+        : { pools: [] };
+
+      const missingPoolStateMaps: PoolStateMap = {};
+      missingPoolStates.pools.forEach(
+        pool => (missingPoolStateMaps[pool.id.toLowerCase()] = pool),
+      );
```

Comment:
I don't think you have a case where the pools will be missing. This can be removed

Reply:
Removed